Reworked INTRODUCTION.md (#10)
ove/ove-docs: docs/INTRODUCTION.md

# The Open Visualisation Environment (OVE)

OVE was developed to meet the requirements of controlling the [Data Observatory](https://www.imperial.ac.uk/data-science/data-observatory/) at the [Data Science Institute](https://www.imperial.ac.uk/data-science/) of [Imperial College London](https://www.imperial.ac.uk), but it is not specialized for that purpose.

OVE can be used for visual analytics on Large High Resolution Displays, for presentations, or for collaborative group work. It allows a user to control the display of content in web browsers distributed across multiple computers by implementing a microservices architecture that allows the distributed execution of applications using web technologies.

The main components of OVE are [**OVE Core**](https://github.com/ove/ove), which controls sections and the applications running within them and [**OVE Apps**](https://github.com/ove/ove-apps), which provide a set of useful applications for common tasks such as displaying images, displaying webpages, and drawing graphs.

> [**OVE Services**](https://github.com/ove/ove-services) provides core functionality microservices within OVE such as the layout service. [**OVE Asset Services**](https://github.com/ove/ove-asset-services) provides an Asset Manager along with a collection of Asset Processing Services, used to upload archives or divide large networks or images into tiles, and serve these to the corresponding applications.

Reviewer: A user who doesn't knwo what **OVE Services** are probably also has no idea what "the layout service" is.

> OVE also provides [graphical editors](https://github.com/ove/ove-editor) and [software development kits](https://github.com/ove/ove-sdks) that can be used to design and develop visualisation projects.

Reviewer: There is currently only a single graphical editor

> OVE also provides [graphical editors](https://github.com/ove/ove-editor) and [software development kits](https://github.com/ove/ove-sdks) that can be used to design and develop visualisation projects.

Reviewer: "visualisation projects" - projects aren't necessarily visualizations, which is why we decided to call them *projects* rather than *visualizations*

## Basic Concepts

> An OVE `server` installation supports multiple `spaces`. A `space` is a collection of monitors, which may be attached to different computers, that together form a single display. OVE is designed to be used in Large High Resolution Display environments (such as a [Data Observatory](https://www.imperial.ac.uk/data-science/data-observatory/) that is 2.53m x 6.00m in size and 30,720 x 4,320 in resolution, consisting of 64 monitors). But, OVE is also suitable for use on much smaller displays.

Reviewer: I think it is useful to communicate that an OVE installation could consist of a single monitor.

> An OVE `server` installation supports multiple `spaces`. A `space` is a collection of monitors, which may be attached to different computers, that together form a single display. OVE is designed to be used in Large High Resolution Display environments (such as a [Data Observatory](https://www.imperial.ac.uk/data-science/data-observatory/) that is 2.53m x 6.00m in size and 30,720 x 4,320 in resolution, consisting of 64 monitors). But, OVE is also suitable for use on much smaller displays.

Reviewer: **the** Data Observatory

**which is**

Author: There was no need to be specific about the Imperial DO, hence this language. Removing it altogether to avoid this disconnection.

> In each space, OVE runs within a number of `clients`. An OVE `client` is a browser window and typically runs full-screen on a single `screen`. A `screen` can be a single monitor or can span multiple monitors if they were all attached to a single computer. The arrangement of `clients` is described in the [`Spaces.json`](https://github.com/dsi-icl/ove/blob/master/packages/ove-core/src/client/Spaces.json) file.

Reviewer: ```suggestion
In each space, OVE runs within a number of `clients`. An OVE `client` is a browser window and typically runs full-screen on a single `screen`. A `screen` can be a single monitor or can span multiple monitors that are attached to the same computer. The arrangement of `clients` is described in the [`Spaces.json`](https://github.com/dsi-icl/ove/blob/master/packages/ove-core/src/client/Spaces.json) file.
```

Author: Isn't it just two ways of saying the same thing? Is there any difference in meaning?

Reviewer: - the tense of 'if they were' and 'they are' is different

- I think that 'the same computer' is clearer than 'a single computer', because there may be many computers, each with more than one monitor attached (whereas 'all attached to a single computer' could suggest that this only applies when the whole installation is controlled by a single computer)

Author: Not convinced on these points, James. I still believe that both versions would be interpreted by the reader in the same way, and as discussed offline, we can argue that either variant is more relevant in its own context. I would therefore stick to the original wording.

> In each space, OVE runs within a number of `clients`. An OVE `client` is a browser window and typically runs full-screen on a single `screen`. A `screen` can be a single monitor or can span multiple monitors if they were all attached to a single computer. The arrangement of `clients` is described in the [`Spaces.json`](https://github.com/ove/ove/blob/master/packages/ove-core/src/client/Spaces.json) file.

Reviewer: ```suggestion
In each space, OVE runs within a number of `clients`. An OVE `client` is a browser window and typically runs full-screen on a single `screen`. A `screen` can span a single monitor or can span multiple monitors that are attached to the same computer. The arrangement of `clients` is described in the [`Spaces.json`](https://github.com/ove/ove/blob/master/packages/ove-core/src/client/Spaces.json) file.
```

Reviewer: Susan agrees that "that are attached to the same computer" is preferable. She also pointed out that the `screen` is not actually 'a single monitor', but rather is **on** it.

Author: I do accept the typo WRT `span` instead of `be`. 

But, there is no gain in debating over the two alternatives, :) for the rest of this line. They would mean the same thing to the reader. I have chosen your alternative, and ending this debate, along the lines of productive use of time.

An OVE `application` (or `app`) includes server-side and client-side components. Each [OVE App](https://github.com/ove/ove-apps/packages) confirm to a [common structure](./APP_DEVELOPMENT.md/#application-structure). Each of the [OVE Apps](https://github.com/ove/ove-apps) provide a way to display a distinct form of commonly used content (subject to the limitations described in the [list of pitfalls to avoid](./PITFALLS.md)).

Each individual `instance` of an OVE `app` is designed to run within its own `section`. `Sections` may span multiple `clients` and can overlap. `Sections` are rectangular regions within an `oveCanvas`. Each `section` has its own `geometry` which is a combination of its `height`, `width`, and 2D coordinates (`x`, `y`). The `geometry` describes the region on which a `section` is deployed on an `oveCanvas`. An `oveCanvas` can also have `groups` of one or more `sections` and `groups`.

> Each individual OVE `app` can have its own `configuration` (or `config`) defined within a [config.json](./APP_DEVELOPMENT.md/#application-structure). The `config` is used to define named `states` other app-specific configuration, which are common to all app `instances`. Each individual `instance` of an OVE `app` can have its own `state` that can be accessed by providing the `id` of the `section` in which the `app` is deployed.

Reviewer: "can have" -> "has"

Author: This is accurate: config.json is optional. See https://github.com/ove/ove/blob/master/packages/ove-lib-appbase/src/index.js#L27.

> Each individual OVE `app` can have its own `configuration` (or `config`) defined within a [config.json](./APP_DEVELOPMENT.md/#application-structure). The `config` is used to define named `states` other app-specific configuration, which are common to all app `instances`. Each individual `instance` of an OVE `app` can have its own `state` that can be accessed by providing the `id` of the `section` in which the `app` is deployed.

Reviewer: By providing the `id` to what?

OVE can be used for visual analytics on Large High Resolution Displays, for presentations, or for collaborative group work. It allows a user to control the display of content in web browsers distributed across multiple computers by implementing a microservices architecture that allows the distributed execution of applications using web technologies.

The main components of OVE are [**OVE Core**](https://github.com/ove/ove), which controls sections and the applications running within them and [**OVE Apps**](https://github.com/ove/ove-apps), which provide a set of useful applications for common tasks such as displaying webpages, images or videos and drawing graphs.

> [**OVE Services**](https://github.com/ove/ove-services) provides core functionality microservices within OVE such as the [Layout Service](https://github.com/ove/ove-services/tree/master/packages/ove-service-layout), which is used to compute absolute positions in pixels from positions expressed relative to a grid or as a percentage of the total space size. [**OVE Asset Services**](https://github.com/ove/ove-asset-services) provides an Asset Manager along with a collection of Asset Processing Services, used to upload archives or divide large networks or images into tiles, and serve these to the corresponding applications.

Reviewer: *which are* used

Author: Existing text is accurate: The `Layout Service` is not plural.

Reviewer: I meant in the second sentence ("an Asset Manager along with a collection of Asset Processing Services, ...")

Author: ok, that wasn't obvious, :). I have fixed this in 40bf207.
